src/rt/util/typeinfo.d

```diff
@@ -20,25 +21,21 @@ if (is(T == float) || is(T == double) || is(T == real))
 
     int compare(T d1, T d2)
     {
-        if (d1 != d1 || d2 != d2) // if either are NaN
+        auto d = d1 - d2;
+        auto result = int(d > 0) - int(d < 0);
+        if (!result && d != d) // if either are NaN
         {
-            if (d1 != d1)
-            {
-                if (d2 != d2)
-                    return 0;
-                return -1;
-            }
-            return 1;
+            return int(d1 == d1) - int(d2 == d2);
         }
-        return (d1 == d2) ? 0 : ((d1 < d2) ? -1 : 1);
+        return result;
```

Comment:
Is this refactoring objectively better? If it isn't, I find it significantly more difficult to understand.

Reply:
it's supposed to be faster